artifactory/commands/flexpack/gradle.go

```diff
@@ -0,0 +1,955 @@
+	if _, ok := props["rootDir"]; !ok {
+		props["rootDir"] = workingDir
+	}
+	if _, ok := props["projectDir"]; !ok {
+		props["projectDir"] = workingDir
+	}
+
+	if repo, err := findRepoInProperties(props, isSnapshot); err == nil && repo != "" {
+		log.Debug("Found repository from properties: " + repo)
+		return repo, nil
+	}
+
+	buildGradlePath := filepath.Join(workingDir, "build.gradle")
+	isBuildGradleKts := false
+	if _, err := os.Stat(buildGradlePath); os.IsNotExist(err) {
+		buildGradlePath = filepath.Join(workingDir, "build.gradle.kts")
+		isBuildGradleKts = true
+	}
+
+	if content, err := os.ReadFile(buildGradlePath); err == nil {
+		if repo, err := findRepoInGradleScript(content, isBuildGradleKts, props, isSnapshot, buildGradlePath); err == nil && repo != "" {
+			log.Debug("Found repository from publishing configuration in " + filepath.Base(buildGradlePath) + ": " + repo)
+			return repo, nil
+		}
+	}
+
+	// Check settings.gradle or settings.gradle.kts
+	settingsGradlePath := filepath.Join(workingDir, "settings.gradle")
+	isSettingsGradleKts := false
+	if _, err := os.Stat(settingsGradlePath); os.IsNotExist(err) {
+		settingsGradlePath = filepath.Join(workingDir, "settings.gradle.kts")
+		isSettingsGradleKts = true
+	}
+
+	if content, err := os.ReadFile(settingsGradlePath); err == nil {
+		if repo, err := findRepoInGradleScript(content, isSettingsGradleKts, props, isSnapshot, settingsGradlePath); err == nil && repo != "" {
+			log.Debug("Found repository from " + filepath.Base(settingsGradlePath) + ": " + repo)
+			return repo, nil
+		}
+	}
+
+	// Check init scripts in GRADLE_USER_HOME
+	gradleUserHome := getGradleUserHome()
+	if gradleUserHome != "" {
+		if repoKey, err := checkInitScripts(gradleUserHome, isSnapshot, props); err == nil && repoKey != "" {
+			log.Debug("Found repository from init scripts: " + repoKey)
+			return repoKey, nil
+		}
+	}
+
+	return "", fmt.Errorf("no deployment repository found in Gradle configuration or environment")
+}
+
+func collectAllGradleProperties(workingDir string) map[string]string {
+	props := make(map[string]string)
+
+	// Helper to merge maps (later sources override earlier ones)
+	merge := func(source map[string]string) {
+		for k, v := range source {
+			if v != "" {
+				props[k] = v
+			}
+		}
+	}
+
+	// 1. GRADLE_USER_HOME gradle.properties (lowest priority)
+	if home := getGradleUserHome(); home != "" {
+		merge(readPropertiesFile(filepath.Join(home, "gradle.properties")))
+	}
+
+	// 2. Project gradle.properties
+	propsFile := filepath.Join(workingDir, "gradle.properties")
+	if _, err := os.Stat(propsFile); err == nil {
+		merge(readPropertiesFile(propsFile))
+	}
+
+	// 3. get project properties from environment variables
+	for _, env := range os.Environ() {
+		if strings.HasPrefix(env, "ORG_GRADLE_PROJECT_") {
+			parts := strings.SplitN(env, "=", 2)
+			if len(parts) == 2 {
+				// the length of that prefix string is exactly 19 characters:
+				key := strings.TrimSpace(parts[0][19:])
```

Comment:
why is the magic number 19?


```suggestion
				key := strings.TrimSpace(parts[0][prefixLen:])
```